src/choice_intf.ml

```diff
@@ -35,6 +35,10 @@ module type Complete = sig
 
   val with_thread : (thread -> 'b) -> 'b t
```

Comment:
is this still used after your PR ?

Reply:
Yes, I may try to delete it in a next PR